### DISCUSSION TITLE
Array query arguments expecting type definition

### DISCUSSION THREAD

**Discussion by henser** [webonyx/graphql-php#1080] [Q&A]
Hi there 👋

First of all, big thanks to all the community involved in contributing to this GraphQL PHP implementation! 🙌

I'm raising up this issue as I've come across a recurring error in my Webonyx GraphQL PHP integration related to query arguments types.

When a client passes an array argument to a query as shown in the example below:

```
{
  myQuery(argument: ["someValue"]) {
    property
  }
}
```
this produces the following type exception:

`GraphQL\Type\Definition\Type::getNullableType(): Argument #1 ($type) must be of type GraphQL\Type\Definition\Type, null given, called in webonyx/graphql-php/src/Validator/Rules/ValuesOfCorrectType.php on line 75`

resulting in the request returning a 200 status code with an empty response.

Performing the same query with a boolean, integer, string, object or null argument value does not produce the same error and runs as expected.

Should this be handled differently where we allow for array type arguments even if they're not defining a type, but simply holding array values? If not possible or intended by design, would it make sense to adjust the error handling in this case so the request returns a readable and contextual error?

**Comment by spawnia** [comment]
What does the schema definition for `myQuery` look like?

**Reply by henser** [reply]
Hej, @spawnia! Thx for reaching back so swiftly. This issue is really not depending on the schema definition as it happens for values of undefined arguments as well. Looking a bit further into it I noticed that this issue was actually tackled by @jdfadams back in July 26 last year [(commit a3760a6)](https://github.com/webonyx/graphql-php/commit/a3760a69d2d341a19c5f2085e219c175b81a259a) in [this PR](https://github.com/webonyx/graphql-php/pull/895). More specifically, the location is `Validator/Rules/ValuesOfCorrectType.php:75` and the issue is that `Type::getNullableType()` expects a `Type` attribute where `$context->getParentInputType()` is passed as its attribute but can be `null`.
Now, the weird thing here is that I don't have that change in my current webonyx/graphql-php package install and I'm currently running the latest v14.11.5 🤔

**Reply by henser** [reply]
After some further digging into it and trying out new and old versions I realized I only get the latest code if I install the package based on the `dev-master` branch. So it seems that the version specific installs like `v14.11.0` or `v14.11.5` or the updated versions like `^v14.9.0` don't really work properly. This suggests that something may be going wrong with the package releases ...

**Reply by spawnia** [reply]
Nah, we are just working towards v15 and have not released in a while.

**Reply by henser** [reply]
ok, I see the 14.11.5 release package here from Jan 24 but so these releases don't get reflected when installed via composer? Is 14.9.0 the last composer available package release? Do you have an estimation when the new v15 will be released and available?

**Reply by spawnia** [reply]
14.x is a separate branch. v15 will take a few weeks, but you can use `dev-master` in the meantime. It is well tested.

**Reply by henser** [reply]
cool! thanks for the info 👍 Got quite confused on the release package contents there for a while but I get it now 😅 🙌

**Reply by tim-hanssen** [reply]
Hey @spawnia any idea on when v15 will be released? If not I will update to the current master but that has some breaking changes so we need to find them.

**Reply by spawnia** [reply]
I have no timeline for that.
